sls-versions/src/main/java/com/palantir/sls/versions/OrderableSlsVersion.java

```diff
@@ -51,32 +44,55 @@ public static Optional<OrderableSlsVersion> safeValueOf(String value) {
             return Optional.empty();
         }
 
-        for (SlsVersionType type : ORDERED_VERSION_TYPES) {
-            MatchResult groups = type.getParser().tryParse(value);
-            if (groups != null) {
-                return Optional.of(construct(type, value, groups));
-            }
+        MatchResult groups = SlsVersionType.RELEASE.getParser().tryParse(value);
```

Comment:
I'm not particularly fond of doing multiple parsings, when we could trivially extend the `ReleaseVersionParser` to parse everything for us (and be more efficient too).

I'm going to look into that in a separate follow-up PR